app/controllers/topics_controller.rb

```diff
@@ -8,6 +8,38 @@ def index
     end
   end
 
+  def vote
+    @topic = Topic.find params[:id]
+    @topic.increment :votes
+    respond_to do |format|
+      if @topic.save 
```

Comment:
Yes, I would make it unsigned! That way you don't need to validate it in Ruby, and also our topics can be voted on more than 9223372036854775807 times (clearly necessary for some of my topics).